packages/internals/src/utils/cloudflareD1.ts

```diff
@@ -0,0 +1,47 @@
+import path from 'node:path'
+import process from 'node:process'
+
+import glob from 'globby'
+import { match } from 'ts-pattern'
+
+const defaultD1DirPath = path.join('.wrangler', 'state', 'v3', 'd1', 'miniflare-D1DatabaseObject')
+
+type TocateLocalCloudflareD1Args = {
+  arg: '--to-local-d1' | '--from-local-d1'
+}
+
+// Utility to find the location of the local Cloudflare D1 sqlite database.
+// When using `wrangler`, the database is located in `${cwd}/.wrangler/state/v3/d1/miniflare-D1DatabaseObject/<UUID>.sqlite`,
+// where `<UUID>` is a unique identifier for the database.
+export async function locateLocalCloudflareD1({ arg }: TocateLocalCloudflareD1Args) {
+  const cwd = process.cwd()
+  const d1DirPath = path.join(cwd, defaultD1DirPath)
+
+  const d1Databases = await glob(path.join(d1DirPath, '*.sqlite'), {})
+
+  if (d1Databases.length === 0) {
+    throw new Error(
+      `No Cloudflare D1 databases found in ${defaultD1DirPath}. Did you run \`wrangler d1 create <DATABASE_NAME>\` and \`wrangler dev\`?`,
```

Comment:
I had originally taken the `wrangler dev` bit from https://github.com/prisma/prisma/pull/23613.
However, that doesn't seem to be enough.

The minimum effort required to generate a local D1 database on disk (besides `wrangler d1 create __YOUR_DATABASE_NAME__`) is to run at least a query on such local database. For example, we can ask users to run:

```
npx wrangler d1 execute __YOUR_DATABASE_NAME__ --local --command "PRAGMA table_list"
```

which shows:

```
 ⛅️ wrangler 3.39.0 (update available 3.42.0)
-------------------------------------------------------
🌀 Mapping SQL input into an array of statements
🌀 Executing on local database __YOUR_DATABASE_NAME__ (1cf2f856-fefd-46e7-a888-f192a6edc6aa) from .wrangler/state/v3/d1:
🌀 To execute on your remote database, add a --remote flag to your wrangler command.
┌────────┬────────────────────┬───────┬──────┬────┬────────┐
│ schema │ name               │ type  │ ncol │ wr │ strict │
├────────┼────────────────────┼───────┼──────┼────┼────────┤
│ main   │ _cf_KV             │ table │ 2    │ 1  │ 0      │
├────────┼────────────────────┼───────┼──────┼────┼────────┤
│ main   │ sqlite_schema      │ table │ 5    │ 0  │ 0      │
├────────┼────────────────────┼───────┼──────┼────┼────────┤
│ temp   │ sqlite_temp_schema │ table │ 5    │ 0  │ 0      │
└────────┴────────────────────┴───────┴──────┴────┴────────┘
```

Reply:
Note: we can still merge this as is. As @nikolasburk said, we can assume that the local database already exists when our users try running `migrate diff --from-local-d1` or `--to-local-d1`.